pkgs/by-name/vs/vsmartcard-vpcd/package.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  libtool,
+  autoreconfHook,
+  pcsclite,
+  qrencode,
+  python3,
+  help2man,
+  darwin,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vsmartcard-vpcd";
+  version = "0.9-unstable-2025-01-25";
+
+  src = fetchFromGitHub {
+    owner = "frankmorgner";
+    repo = "vsmartcard";
+    rev = "7369dae26bcb709845003ae2128b8db9df7031ae";
+    hash = "sha256-mfw/Yv12ceBVZIyAKJqBh+w4otj3rYYZbJUjKRLcsr4=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/virtualsmartcard";
+
+  nativeBuildInputs = [
+    autoreconfHook
+    libtool
+    pkg-config
+    help2man
+  ];
+
+  buildInputs =
+    [
+      pcsclite
+      qrencode
+      (python3.withPackages (
+        pp: with pp; [
+          pyscard
+          pycrypto
+          pbkdf2
+          pillow
+          gnureadline
+        ]
+      ))
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.PCSC
+    ];
+
+  configureFlags = lib.optional stdenv.isDarwin "--enable-infoplist";
+
+  meta = {
+    description = "Emulates a smart card and makes it accessible through PC/SC";
+    homepage = "http://frankmorgner.github.io/vsmartcard/virtualsmartcard/README.html";
+    license = lib.licenses.gpl3Only;
+    platforms = lib.platforms.all;
```

Doesn't build on macOS:
```
  CCLD     libifdvpcd.la
Undefined symbols for architecture arm64:
  "_log_msg", referenced from:
      _IFDHCreateChannel in libifdvpcd_la-ifd-vpcd.o
      _IFDHCreateChannelByName in libifdvpcd_la-ifd-vpcd.o
      _IFDHControl in libifdvpcd_la-ifd-vpcd.o
      _IFDHCloseChannel in libifdvpcd_la-ifd-vpcd.o
      _IFDHGetCapabilities in libifdvpcd_la-ifd-vpcd.o
      _IFDHSetCapabilities in libifdvpcd_la-ifd-vpcd.o
      _IFDHSetProtocolParameters in libifdvpcd_la-ifd-vpcd.o
      ...
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```suggestion
    platforms = lib.platforms.all;
    broken = stdenv.isDarwin;
```

---

Okay, I'll mark Darwin as broken for now. I  considered setting platforms to Linux only, since I don't have a Darwin system to test. But the upstream supports it so I'll leave it for interested contributors to jump off from.